app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -53,26 +49,23 @@ provideFunctionInfo(
   }),
 )
 
-const methodCallInfo = computed(() => {
-  return getMethodCallInfoRecursively(props.input.value, graph.db)
-})
-
-const interpreted = computed(() => {
-  return interpretCall(props.input.value, methodCallInfo.value == null)
-})
+const { methodCallInfo, interpreted, visualizationConfig, selfArgumentPreapplied, subjectInfo } =
```

Comment:
I think this API could be simplified a bit.

The responsibility of `useWidgetFunctionCallInfo` is to analyze AST and ExpressionUpdates data; however, fully interpreting ExpressionUpdates requires some dynamicConfig information. Because the composable doesn't have this information, some of its outputs seem like intermediate values that require some context to understand.

If the composable were to take `widgetConfiguration` as an additional argument, it could return `application` instead of `interpreted`, `selfArgumentPreapplied`, and `subjectInfo`.